python/pyarrow/tests/test_flight.py

```diff
@@ -452,6 +453,17 @@ def do_action(self, context, action):
         return iter([flight.Result(b"")])
 
 
+class MultiHeaderFlightServer(FlightServerBase):
+    """Test sending/receiving multiple (binary-valued) headers."""
+
+    def do_action(self, context, action):
+        middleware = context.get_middleware("test")
+        if middleware:
+            headers = repr(middleware.client_headers).encode("utf-8")
+            return iter([flight.Result(headers)])
```

Comment:
Is `iter` required?

Reply:
It was before but I've changed it now so that it's not.